### PR TITLE
[UIKit] Update bindings to Xcode 12 Beta 2

### PR DESCRIPTION
**Pull request by dalexsoto**
We are still missing some APIs that can't be bound in this PR, the following task items are not necessarily needed to merge this PR but will need to be completed by Xcode 12 final:

- Our generator cannot create block trampolines with generic NSObject subclasses. Filled issue https://github.com/xamarin/xamarin-macios/issues/8993
    - [ ] Enable `UICollectionViewDiffableDataSourceSectionSnapshotHandlers<ItemType>` commented properties.
    - [ ] Enable `UICollectionViewDiffableDataSourceReorderingHandlers<SectionType, ItemType>` commented properties.
- ~~`UTType` from UniformTypeIdentifiers framework is not bound yet:~~
    - [x] Enable `UIDocumentPickerViewController` constructors when `UTType` is bound.
    - [x] Enable `UIDocumentBrowserViewController.ctor` when `UTType` is bound.
    - [x] Enable `UIDocumentBrowserViewController.ContentTypesForRecentDocuments` when `UTType` is bound.
- `NSOrderedCollectionDifference` from Foundation is not bound yet:
    - [ ] Enable `NSDiffableDataSourceSectionTransaction<SectionIdentifierType, ItemIdentifierType>.Difference`
    - [ ] Enable `NSDiffableDataSourceTransaction<SectionIdentifierType, ItemIdentifierType>.Difference`
- Expose `Intents` framework to tvOS
    - [ ] Expose `UIApplicationDelegate.GetHandlerForIntent` to tvOS
- [NoWatch, NoTV, NoiOS] ?
    - [ ] Check future availability on `UICollectionView.ContextMenuInteraction` [NoWatch, NoTV, NoiOS]
    - [ ] Check future availability on `UISceneActivationRequestOptions.CollectionJoinBehavior` [NoWatch, NoTV, NoiOS]

The above items are now part of https://github.com/xamarin/xamarin-macios/issues/8943

### CODE REVIEW

File: src/UIKit/UICellAccessory.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	static internal class SDUICellAccessoryPosition {
+		static internal readonly DUICellAccessoryPosition Handler = Invoke;
+		
+		[MonoPInvokeCallback (typeof (DUICellAccessoryPosition))]
+		static unsafe nuint Invoke (IntPtr block, IntPtr accessories) {
+			var descriptor = (BlockLiteral *) block;
+			var del = (UICellAccessoryPosition) (descriptor->Target);
+			nuint retval = del (NSArray.ArrayFromHandle<UICellAccessory> (accessories));
+			return retval;
+		}
```
Comment (mandel-macaque) [inline src/UIKit/UICellAccessory.cs:79]: I believe you can use BlockLiteralRef and avoid it from being unsafe, something like:
```csharp
static unsafe nuint Invoke (ref BlockLiteral block, IntPtr accessories) 
```

File: src/UIKit/UICellAccessory.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	} /* class SDUICellAccessoryPosition */
+
+	internal sealed class NIDUICellAccessoryPosition : TrampolineBlockBase {
+		DUICellAccessoryPosition invoker;
+		
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public unsafe NIDUICellAccessoryPosition (BlockLiteral *block) : base (block)
+		{
+			invoker = block->GetDelegateForBlock<DUICellAccessoryPosition> ();
+		}
```
Comment (mandel-macaque) [inline src/UIKit/UICellAccessory.cs:89]: Same, can we use ref?

File: src/UIKit/UICellAccessory.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+		
+		[Preserve (Conditional = true)]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public unsafe static UICellAccessoryPosition? Create (IntPtr block)
+		{
+			if (block == IntPtr.Zero)
+				return null;
+			var del = (UICellAccessoryPosition) GetExistingManagedDelegate (block);
+			return del ?? new NIDUICellAccessoryPosition ((BlockLiteral *) block).Invoke;
```
Comment (mandel-macaque) [inline src/UIKit/UICellAccessory.cs:98]: Move unsafe inside with an unsafe block:
```suggestion
			var del = (UICellAccessoryPosition) GetExistingManagedDelegate (block);
			unsafe {
			   return del ?? new NIDUICellAccessoryPosition ((BlockLiteral *) block).Invoke;
			}
```

File: src/UIKit/UICellAccessory.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		}
+		
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		unsafe nuint Invoke (UICellAccessory [] accessories)
```
Comment (mandel-macaque) [inline src/UIKit/UICellAccessory.cs:102]: In unsafe needed?

File: src/UIKit/UIConfigurationColorTransformer.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	internal delegate IntPtr DUIConfigurationColorTransformerHandler (IntPtr block, IntPtr color);
+
+	//
+	// This class bridges native block invocations that call into C#
+	//
+	static internal class SDUIConfigurationColorTransformerHandler {
+		static internal readonly DUIConfigurationColorTransformerHandler Handler = Invoke;
+		
+		[MonoPInvokeCallback (typeof (DUIConfigurationColorTransformerHandler))]
+		static unsafe IntPtr Invoke (IntPtr block, IntPtr color) {
```
Comment (mandel-macaque) [inline src/UIKit/UIConfigurationColorTransformer.cs:40]: ref BlockLiteral? We get unsafe out.

File: src/UIKit/UIConfigurationColorTransformer.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+			invoker = block->GetDelegateForBlock<DUIConfigurationColorTransformerHandler> ();
+		}
+		
+		[Preserve (Conditional = true)]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public unsafe static UIConfigurationColorTransformerHandler? Create (IntPtr block)
+		{
+			if (block == IntPtr.Zero)
+				return null;
+			var del = (UIConfigurationColorTransformerHandler) GetExistingManagedDelegate (block);
```
Comment (mandel-macaque) [inline src/UIKit/UIConfigurationColorTransformer.cs:63]: Same, remove unsafe from the signature, use an usafe block.

File: src/UIKit/UIConfigurationColorTransformer.cs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+			return del ?? new NIDUIConfigurationColorTransformerHandler ((BlockLiteral *) block).Invoke;
+		}
+		
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		unsafe UIColor Invoke (UIColor color)
```
Comment (mandel-macaque) [inline src/UIKit/UIConfigurationColorTransformer.cs:68]: Do we need the unsafe?

File: src/appkit.cs [hunk 8]
```diff
@@ -10535,6 +10535,18 @@ interface NSLayoutXAxisAnchor
...
+		[Export ("constraintEqualToSystemSpacingAfterAnchor:multiplier:")]
+		NSLayoutConstraint ConstraintEqualToSystemSpacingAfterAnchor (NSLayoutXAxisAnchor anchor, nfloat multiplier);
+
+		[Mac (10,16)]
+		[Export ("constraintGreaterThanOrEqualToSystemSpacingAfterAnchor:multiplier:")]
+		NSLayoutConstraint ConstraintGreaterThanOrEqualToSystemSpacingAfterAnchor (NSLayoutXAxisAnchor anchor, nfloat multiplier);
+
+		[Mac (10,16)]
+		[Export ("constraintLessThanOrEqualToSystemSpacingAfterAnchor:multiplier:")]
+		NSLayoutConstraint ConstraintLessThanOrEqualToSystemSpacingAfterAnchor (NSLayoutXAxisAnchor anchor, nfloat multiplier);
```
Comment (mandel-macaque) [inline src/appkit.cs:10549]: I understand the naming.. but is so verbose! do we need, I'd do:
```suggestion
		NSLayoutConstraint EqualToSystemSpaceAfterAnchor (NSLayoutXAxisAnchor anchor, nfloat multiplier);

		[Mac (10,16)]
		[Export ("constraintGreaterThanOrEqualToSystemSpacingAfterAnchor:multiplier:")]
		NSLayoutConstraint GreaterThanOrEqualToSystemSpaceAfterAncho (NSLayoutXAxisAnchor anchor, nfloat multiplier);

		[Mac (10,16)]
		[Export ("constraintLessThanOrEqualToSystemSpacingAfterAnchor:multiplier:")]
		NSLayoutConstraint LessThanOrEqualToSystemSpaceAfterAncho (NSLayoutXAxisAnchor anchor, nfloat multiplier);
```

File: src/appkit.cs [hunk 9]
```diff
@@ -10545,6 +10557,18 @@ interface NSLayoutYAxisAnchor
...
+		[Export ("constraintEqualToSystemSpacingBelowAnchor:multiplier:")]
+		NSLayoutConstraint ConstraintEqualToSystemSpacingBelowAnchor (NSLayoutYAxisAnchor anchor, nfloat multiplier);
+
+		[Mac (10,16)]
+		[Export ("constraintGreaterThanOrEqualToSystemSpacingBelowAnchor:multiplier:")]
+		NSLayoutConstraint ConstraintGreaterThanOrEqualToSystemSpacingBelowAnchor (NSLayoutYAxisAnchor anchor, nfloat multiplier);
+
+		[Mac (10,16)]
+		[Export ("constraintLessThanOrEqualToSystemSpacingBelowAnchor:multiplier:")]
+		NSLayoutConstraint ConstraintLessThanOrEqualToSystemSpacingBelowAnchor (NSLayoutYAxisAnchor anchor, nfloat multiplier);
```
Comment (mandel-macaque) [inline src/appkit.cs:10571]: Same, super verbose:
```suggestion

		[Mac (10,16)]
		[Export ("constraintEqualToSystemSpacingBelowAnchor:multiplier:")]
		NSLayoutConstraint EqualToSystemSpacingBelowAnchor (NSLayoutYAxisAnchor anchor, nfloat multiplier);

		[Mac (10,16)]
		[Export ("constraintGreaterThanOrEqualToSystemSpacingBelowAnchor:multiplier:")]
		NSLayoutConstraint GreaterThanOrEqualToSystemSpacingBelowAnchor (NSLayoutYAxisAnchor anchor, nfloat multiplier);

		[Mac (10,16)]
		[Export ("constraintLessThanOrEqualToSystemSpacingBelowAnchor:multiplier:")]
		NSLayoutConstraint LessThanOrEqualToSystemSpacingBelowAnchor (NSLayoutYAxisAnchor anchor, nfloat multiplier);
```

File: src/uikit.cs [hunk 10]
```diff
@@ -3253,12 +3282,24 @@ interface UIContextMenuInteraction : UIInteraction {
...
 		[Export ("initWithDelegate:")]
 		[DesignatedInitializer]
 		IntPtr Constructor (IUIContextMenuInteractionDelegate @delegate);
 
 		[Export ("locationInView:")]
 		CGPoint GetLocation ([NullAllowed] UIView inView);
+
+		[iOS (14,0)]
+		[Export ("updateVisibleMenuWithBlock:")]
+		void UpdateVisibleMenu (Func<UIMenu, UIMenu> handler);
```
Comment (mandel-macaque) [inline src/uikit.cs:3298]: Shall we use a delegate with names of what UIMenu first and UIMenu second are?

File: src/uikit.cs [hunk 11]
```diff
@@ -11254,8 +11553,12 @@ interface UISearchDisplayDelegate {
 	[BaseType (typeof (NSObject))]
 	partial interface UISearchResultsUpdating {
 		[Abstract]
-	    [Export ("updateSearchResultsForSearchController:")]
-	    void UpdateSearchResultsForSearchController (UISearchController searchController);
+		[Export ("updateSearchResultsForSearchController:")]
+		void UpdateSearchResultsForSearchController (UISearchController searchController);
```
Comment (mandel-macaque) [inline src/uikit.cs:11557]: Do we need 'ForSearchController'?

File: src/UIKit/UICellAccessory.cs (hunk 4 above)
Comment (dalexsoto) [reply src/UIKit/UICellAccessory.cs:102]: Nope sorry, missed that one cleaning the code produced by the generator

File: src/appkit.cs (hunk 8 above)
Comment (dalexsoto) [reply src/appkit.cs:10549]: Unfortunately we can't do that, maybe for XAMCORE_4_0. This code exist on iOS too using the same names so we can increase code sharing between the platforms.

File: src/appkit.cs (hunk 9 above)
Comment (dalexsoto) [reply src/appkit.cs:10571]: Same we can't :( 

https://github.com/xamarin/xamarin-macios/blob/4ed68336d7ebbc70d6386bc9a5240dc1f1dfa5a3/src/uikit.cs#L401-L411

File: src/uikit.cs (hunk 10 above)
Comment (dalexsoto) [reply src/uikit.cs:3298]: We could but we gain not much from that, what you get in the parameter is the visible menu and the method is already called Update`VisibleMenu` and the return is well a menu too :/

File: src/uikit.cs (hunk 11 above)
Comment (dalexsoto) [reply src/uikit.cs:11557]: We can't change it, it would be a breaking change :) this is not new, looks like I fixed indentation

File: src/uikit.cs [hunk 12]
```diff
@@ -12909,21 +13299,63 @@ interface UITableViewDelegate {
...
-		UILabel TextLabel { get;  }
+		UILabel TextLabel { get; }
 
+		[Deprecated (PlatformName.iOS, 14, 0)]
 		[Export ("detailTextLabel", ArgumentSemantic.Retain)]
-		UILabel DetailTextLabel { get;  }
+		UILabel DetailTextLabel { get; }
 
-		[Export ("contentView", ArgumentSemantic.Retain)]
-		UIView ContentView { get;  }
```
Comment (rolfbjarne) [inline src/uikit.cs:12926]: There are some unnecessary whitespace and code moves here.

File: src/uikit.cs [hunk 13]
```diff
@@ -15542,10 +16002,49 @@ interface UITextContentType {
...
+		[Export ("preferredSplitBehavior", ArgumentSemantic.Assign)]
+		UISplitViewControllerSplitBehavior PreferredSplitBehavior { get; set; }
+
+		[TV (14,0), iOS (14,0)]
+		[Export ("splitBehavior")]
+		UISplitViewControllerSplitBehavior SplitBehavior { get; }
+
+		[TV (14,0), iOS (14,0)]
+		[Export ("setViewController:forColumn:")]
+		void SetViewController ([NullAllowed] UIViewController vc, UISplitViewControllerColumn column);
```
Comment (rolfbjarne) [inline src/uikit.cs:16033]:
```suggestion
		void SetViewController ([NullAllowed] UIViewController viewController, UISplitViewControllerColumn column);
```

File: src/uikit.cs [hunk 14]
```diff
@@ -15681,6 +16200,30 @@ interface UISplitViewControllerDelegate {
 		[iOS (8,0)]
 		[Export ("splitViewController:separateSecondaryViewControllerFromPrimaryViewController:"), DelegateName("UISplitViewControllerGetSecondaryViewController"), DefaultValue(null)]
 		UIViewController SeparateSecondaryViewController (UISplitViewController splitViewController, UIViewController primaryViewController);
+
+		[TV (14,0), iOS (14,0)]
+		[Export ("splitViewController:topColumnForCollapsingToProposedTopColumn:"), DelegateName ("UISplitViewControllerGetTopColumnForCollapsing"), DefaultValueFromArgument ("proposedTopColumn")]
+		UISplitViewControllerColumn GetTopColumnForCollapsing (UISplitViewController svc, UISplitViewControllerColumn proposedTopColumn);
```
Comment (rolfbjarne) [inline src/uikit.cs:16206]:
```suggestion
		UISplitViewControllerColumn GetTopColumnForCollapsing (UISplitViewController splitViewController, UISplitViewControllerColumn proposedTopColumn);
```

File: src/uikit.cs [hunk 15]
```diff
@@ -15681,6 +16200,30 @@ interface UISplitViewControllerDelegate {
...
+
+		[TV (14,0), iOS (14,0)]
+		[Export ("splitViewController:displayModeForExpandingToProposedDisplayMode:"), DelegateName ("UISplitViewControllerGetDisplayModeForExpanding"), DefaultValueFromArgument ("proposedDisplayMode")]
+		UISplitViewControllerDisplayMode GetDisplayModeForExpanding (UISplitViewController svc, UISplitViewControllerDisplayMode proposedDisplayMode);
```
Comment (rolfbjarne) [inline src/uikit.cs:16210]:
```suggestion
		UISplitViewControllerDisplayMode GetDisplayModeForExpanding (UISplitViewController splitViewController, UISplitViewControllerDisplayMode proposedDisplayMode);
```

File: src/uikit.cs [hunk 16]
```diff
@@ -15681,6 +16200,30 @@ interface UISplitViewControllerDelegate {
...
+
+		[TV (14,0), iOS (14,0)]
+		[Export ("splitViewControllerDidCollapse:"), EventArgs ("UISplitViewControllerDidExpandCollapse")]
+		void DidCollapse (UISplitViewController svc);
```
Comment (rolfbjarne) [inline src/uikit.cs:16214]:
```suggestion
		void DidCollapse (UISplitViewController splitViewController);
```

File: src/uikit.cs [hunk 17]
```diff
@@ -15681,6 +16200,30 @@ interface UISplitViewControllerDelegate {
...
+
+		[TV (14,0), iOS (14,0)]
+		[Export ("splitViewControllerDidExpand:"), EventArgs ("UISplitViewControllerDidExpandCollapse")]
+		void DidExpand (UISplitViewController svc);
```
Comment (rolfbjarne) [inline src/uikit.cs:16218]:
```suggestion
		void DidExpand (UISplitViewController splitViewController);
```

File: src/uikit.cs [hunk 18]
```diff
@@ -15681,6 +16200,30 @@ interface UISplitViewControllerDelegate {
...
+
+		[TV (14,0), iOS (14,0)]
+		[Export ("splitViewController:willShowColumn:"), EventArgs ("UISplitViewControllerWillShowHideColumn")]
+		void WillShowColumn (UISplitViewController svc, UISplitViewControllerColumn column);
```
Comment (rolfbjarne) [inline src/uikit.cs:16222]:
```suggestion
		void WillShowColumn (UISplitViewController splitViewController, UISplitViewControllerColumn column);
```

File: src/uikit.cs [hunk 19]
```diff
@@ -15681,6 +16200,30 @@ interface UISplitViewControllerDelegate {
...
+
+		[TV (14,0), iOS (14,0)]
+		[Export ("splitViewController:willHideColumn:"), EventArgs ("UISplitViewControllerWillShowHideColumn")]
+		void WillHideColumn (UISplitViewController svc, UISplitViewControllerColumn column);
```
Comment (rolfbjarne) [inline src/uikit.cs:16226]:
```suggestion
		void WillHideColumn (UISplitViewController splitViewController, UISplitViewControllerColumn column);
```

File: src/uikit.cs [hunk 20]
```diff
@@ -17043,6 +17586,7 @@ partial interface UIUserNotificationAction : NSCopying, NSMutableCopying, NSSecu
 		NSString ResponseTypedTextKey { get; }
 	}
 #else
+#if !XAMCORE_4_0 // No longer present in watchOS 7.0
```
Comment (rolfbjarne) [inline src/uikit.cs:17589]: Maybe we should add a `[Deprecated (...)]` then

File: src/uikit.cs [hunk 21]
```diff
@@ -18919,13 +19500,20 @@ interface UIDocumentBrowserViewController : NSCoding {
 		[Export ("allowsPickingMultipleItems")]
 		bool AllowsPickingMultipleItems { get; set; }
 
+		[Deprecated (PlatformName.iOS, 14, 0, message: "no longer supported.")]
```
Comment (rolfbjarne) [inline src/uikit.cs:19503]:
```suggestion
		[Deprecated (PlatformName.iOS, 14, 0, message: "No longer supported.")]
```

File: src/uikit.cs [hunk 22]
```diff
@@ -21095,4 +21740,1026 @@ interface UIPointerShape : NSCopying {
...
+		[return: NullAllowed]
+		[Export ("iconImage")]
+		UIImage GetIconImage ();
+	}
+
+	[NoWatch, TV (14,0), iOS (14,0)]
+	[Static, Partial]
+	interface UIConfigurationColorTransformer {
+
+		// Manually bound inside UIKit/UIConfigurationColorTransformer.cs
```
Comment (rolfbjarne) [inline src/uikit.cs:22659]: Why was this manually bound?

File: src/uikit.cs [hunk 23]
```diff
@@ -21095,4 +21740,1026 @@ interface UIPointerShape : NSCopying {
...
+	}
+
+	[TV (14,0), NoWatch, NoiOS]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UISearchSuggestionItem : UISearchSuggestion {
+
+		[Static]
+		[Export ("suggestionWithLocalizedSuggestion:")]
+		UISearchSuggestionItem Create (string suggestion);
```
Comment (rolfbjarne) [inline src/uikit.cs:22672]: I wonder if localizedSuggestion is one of those strings that have to stay as an NSString so that they can be localized at runtime?

File: src/UIKit/UICellAccessory.cs [hunk 24]
```diff
@@ -0,0 +1,114 @@
...
+	public partial class UICellAccessory {
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[DllImport (Constants.UIKitLibrary)]
+		static extern IntPtr UICellAccessoryPositionBeforeAccessoryOfClass (IntPtr accessoryCls);
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[return: DelegateProxy (typeof (SDUICellAccessoryPosition))]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public static UICellAccessoryPosition GetPositionBeforeAccessory (Class accessoryClass)
```
Comment (rolfbjarne) [inline src/UIKit/UICellAccessory.cs:29]: No tests for this method.

File: src/UIKit/UICellAccessory.cs [hunk 25]
```diff
@@ -0,0 +1,114 @@
...
+			if (accessoryClass == null)
+				throw new ArgumentNullException ("accessoryClass");
+			var ret = UICellAccessoryPositionBeforeAccessoryOfClass (accessoryClass.Handle);
+			return NIDUICellAccessoryPosition.Create (ret)!;
+		}
+		
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[return: DelegateProxy (typeof (SDUICellAccessoryPosition))]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public static UICellAccessoryPosition GetPositionBeforeAccessory (Type accessoryType) => GetPositionBeforeAccessory (new Class (accessoryType));
```
Comment (rolfbjarne) [inline src/UIKit/UICellAccessory.cs:40]: No tests.

File: src/UIKit/UICellAccessory.cs [hunk 26]
```diff
@@ -0,0 +1,114 @@
...
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[DllImport (Constants.UIKitLibrary)]
+		static extern IntPtr UICellAccessoryPositionAfterAccessoryOfClass (IntPtr accessoryCls);
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[return: DelegateProxy (typeof (SDUICellAccessoryPosition))]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public static UICellAccessoryPosition GetPositionAfterAccessory (Class accessoryClass)
```
Comment (rolfbjarne) [inline src/UIKit/UICellAccessory.cs:49]: No tests.

File: src/UIKit/UICellAccessory.cs [hunk 27]
```diff
@@ -0,0 +1,114 @@
...
+			if (accessoryClass == null)
+				throw new ArgumentNullException ("accessoryClass");
+			var ret = UICellAccessoryPositionAfterAccessoryOfClass (accessoryClass.Handle);
+			return NIDUICellAccessoryPosition.Create (ret)!;
+		}
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
+		[return: DelegateProxy (typeof (SDUICellAccessoryPosition))]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public static UICellAccessoryPosition GetPositionAfterAccessory (Type accessoryType) => GetPositionAfterAccessory (new Class (accessoryType));
```
Comment (rolfbjarne) [inline src/UIKit/UICellAccessory.cs:60]: No tests.

File: src/UIKit/UICellAccessory.cs [hunk 28]
```diff
@@ -0,0 +1,114 @@
...
+	} /* class UICellAccessory */
+
+	[UnmanagedFunctionPointerAttribute (CallingConvention.Cdecl)]
+	[UserDelegateType (typeof (UICellAccessoryPosition))]
+	internal delegate nuint DUICellAccessoryPosition (IntPtr block, IntPtr accessories);
+	
+	//
+	// This class bridges native block invocations that call into C#
```
Comment (rolfbjarne) [inline src/UIKit/UICellAccessory.cs:68]: I wonder how complicated it would be to implement support for generating the supporting code for blocks by adding support in the generator for a `[GenerateBlockCode]` attribute to delegates:

```csharp
[GenerateBlockCode]
delegate nuint UICellAccessoryPosition (UICellAccessory [] accessories);
```

File: src/uikit.cs [hunk 29]
```diff
@@ -3532,10 +3573,18 @@ interface UIApplicationDelegate {
...
 		[NoTV]
 		[iOS (11,0), Watch (4,0)]
 		[Export ("application:handleIntent:completionHandler:")]
 		void HandleIntent (UIApplication application, INIntent intent, Action<INIntentResponse> completionHandler);
+
+		// TODO: enable on tvOS once Intents framework is exposed on tvOS.
+		[iOS (14,0), TV (14,0), Watch (7,0)]
+		[Export ("application:handlerForIntent:")]
+		[return: NullAllowed]
+		NSObject GetHandlerForIntent (UIApplication application, INIntent intent);
```
[the remark this answers is not in the document]
Comment (dalexsoto) [reply src/uikit.cs:3587]: I am not sure about this one, you literally lose the intent of the method specially with the huge amount of methods inside the `UIApplicationDelegate`. Also it replaces `HandleIntent` and `GetHandlerForIntent` is a bit more ~googleable~ bingable and easier to find by the autocompletion engine if you type intent.

File: src/uikit.cs (hunk 20 above)
Comment (dalexsoto) [reply src/uikit.cs:17589]: The class itself is already deprecated this being the reason I did not add it.

File: src/uikit.cs (hunk 22 above)
Comment (dalexsoto) [reply src/uikit.cs:22659]: Because FieldAttribute does not support delegates as return types according to our [docs](https://docs.microsoft.com/en-us/xamarin/cross-platform/macios/binding/binding-types-reference#fieldattribute).

> Properties flagged with the Field attribute can be of the following types:
> 
> - NSString
> - NSArray
> - nint / int / long
> - nuint / uint / ulong
> - nfloat / float
> - double
> - CGSize
> - System.IntPtr
> - Enums

File: src/UIKit/UICellAccessory.cs [hunk 30]
```diff
@@ -0,0 +1,111 @@
...
+using ObjCRuntime;
+
+#nullable enable
+#if !WATCH
+
+namespace UIKit {
+
+	public partial class UICellAccessory {
+
+		[Watch (7,0), TV (14,0), iOS (14,0)]
```
Comment (spouliot) [inline src/UIKit/UICellAccessory.cs:22]: you have a `#if !WATCH` above and `[Watch (7,0)]` attributes
only one is right :)

File: src/UIKit/UIEnums.cs [hunk 31]
```diff
@@ -2908,4 +2943,184 @@ public enum UIPointerEffectTintMode : long {
...
+		Plain,
+		Grouped,
+#if !TVOS
+		[NoTV]
+		InsetGrouped,
+		[NoTV]
+		Sidebar,
+		[NoTV]
+		SidebarPlain,
+#endif
```
Comment (spouliot) [inline src/UIKit/UIEnums.cs:3006]: that `#if` should not be needed

File: src/UIKit/UICellAccessory.cs (hunk 2 above)
Comment (spouliot) [reply src/UIKit/UICellAccessory.cs:90]: that's a copy/paste of generated code and, to remain optimizable, it should follow the exact same pattern

File: src/uikit.cs [hunk 32]
```diff
@@ -11254,8 +11553,12 @@ interface UISearchDisplayDelegate {
...
 	partial interface UISearchResultsUpdating {
 		[Abstract]
-	    [Export ("updateSearchResultsForSearchController:")]
-	    void UpdateSearchResultsForSearchController (UISearchController searchController);
+		[Export ("updateSearchResultsForSearchController:")]
+		void UpdateSearchResultsForSearchController (UISearchController searchController);
+
+		[TV (14,0), NoWatch, NoiOS]
+		[Export ("updateSearchResultsForSearchController:selectingSearchSuggestion:")]
+		void SelectingSearchSuggestion (UISearchController searchController, IUISearchSuggestion searchSuggestion);
```
Comment (spouliot) [inline src/uikit.cs:11561]: wrong part of the selector used for the name
it's an overload to the existing one, with an extra parameter

File: src/uikit.cs (hunk 23 above)
Comment (spouliot) [reply src/uikit.cs:22672]: better safe than sorry

**Comment by spouliot** [comment]
build

**Comment by spouliot** [comment]
test failure is a known, random issue -> https://github.com/xamarin/maccore/issues/1083

File: src/uikit.cs (hunk 22 above)
Comment (rolfbjarne) [reply src/uikit.cs:22707]: Seems like we should add that support for that in the generator at some point, to avoid having to manually write block code (filed: https://github.com/xamarin/xamarin-macios/issues/9055).

File: tests/monotouch-test/UIKit/CellAccessoryTest.cs [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+
+		[Test]
+		public void GetPositionBeforeAccessory ()
+		{
+			TestRuntime.AssertXcodeVersion (12, 0);
+			var cap = UICellAccessory.GetPositionBeforeAccessory (new Class ("UIButton"));
+			Assert.NotNull (cap, "Class/cap");
+			Assert.That (cap (new UICellAccessory [0]), Is.EqualTo (0), "Class/Invoke");
+
+			cap = UICellAccessory.GetPositionBeforeAccessory (new Class (typeof (UICellAccessory)));
```
[the remark this answers is not in the document]
Comment (spouliot) [reply tests/monotouch-test/UIKit/CellAccessoryTest.cs]: yes :)

**Comment by spouliot** [comment]
`Java NullPointerException in log` :(

**Comment by spouliot** [comment]
build
